src/dataEditor/dataEditorClient.ts

```diff
@@ -248,14 +256,30 @@ export class DataEditorClient implements vscode.Disposable {
       assert(this.omegaSessionId.length > 0, 'omegaSessionId is not set')
       addActiveSession(this.omegaSessionId)
 
-      this.contentType = createSessionResponse.hasContentType()
+      data.diskFileSize = data.computedFileSize =
+        createSessionResponse.hasFileSize()
+          ? (createSessionResponse.getFileSize() as number)
+          : 0
+
+      data.type = createSessionResponse.hasContentType()
         ? (createSessionResponse.getContentType() as string)
         : 'unknown'
-      assert(this.contentType.length > 0, 'contentType is not set')
+      assert(data.type.length > 0, 'contentType is not set')
+
+      data.byteOrderMark = createSessionResponse.hasByteOrderMark()
+        ? (createSessionResponse.getByteOrderMark() as string)
+        : 'unknown'
+      assert(data.byteOrderMark.length > 0, 'byteOrderMark is not set')
 
-      this.fileSize = createSessionResponse.hasFileSize()
-        ? (createSessionResponse.getFileSize() as number)
-        : 0
+      data.language = createSessionResponse.hasLanguage()
+        ? (createSessionResponse.getLanguage() as string)
+        : 'unknown'
+      assert(data.language.length > 0, 'language is not set')
```

Comment:
@mbeckerle, here is where the Data Editor is getting the detected language.  This happens at session creation time and is detected by the Ωedit™ server (see: https://github.com/ctc-oss/omega-edit/blob/main/server/scala/serv/src/main/scala/com/ctc/omega_edit/grpc/EditorService.scala#L124).  When the session is created in the extension, it sends a message to the WebView front-end with the details for the display.

Reply:
In this patch so far, you can do a segment via the data profiler.  The profiler will detect the BOM at the beginning of the file and do the character counts in the desired segment on demand.  It doesn't currently re-calculate the language or the file-type, however as these are being done at session creation time right now (ref: https://github.com/ctc-oss/omega-edit/blob/main/server/scala/serv/src/main/scala/com/ctc/omega_edit/grpc/EditorService.scala#L115).

Language detection on a segment ought to be fairly straightforward to add to the data profiler.  We can stop doing it at session creation and do it on demand with the profiler.

For content-type I think we just leave that as-is (done only at session creation time).  If the content-type changes (maybe because we opened an empty file and are creating a file from whole cloth), I can add a refresh widget to refresh the content type based on file updates.

In summary, I agree that we ought to be able to do language detection on a segment and discontinue doing it at session creation time, and for content-type, I think a refresh mechanism will be the best bet.

To support these changes, I'll need to update Ωedit™ and cut a new release for the server-side support.